mods/walls/init.lua

```diff
@@ -25,6 +25,55 @@ walls.register = function(wall_name, wall_desc, wall_texture_table, wall_mat, wa
 		walkable = true,
 		groups = { cracky = 3, wall = 1, stone = 2 },
 		sounds = wall_sounds,
+				after_place_node = function(pos, placer, itemstack, pointed_thing)
+			local pos_under = {x = pos.x, y = pos.y - 1, z = pos.z}
+			local pos_above = {x = pos.x, y = pos.y + 1, z = pos.z}
+			local node_under = string.gsub(minetest.get_node(pos_under).name, "_full$", "")
```

Comment:
I don't like the required use of ~regex~ patterns here. As the *walls* group is in the full version, you don't need the canonical version first.

I think this would be better:

```lua
local node_under = minetest.get_node(pos_under).name
if minetest.get_item_group(node_under, "wall") == 1 then
	local node_under_can = string.gsub(node_under, "_full$", "")
	minetest.set_node(pos_under, {name = node_under_can .. "_full"})
end

local node_above = minetest.get_node(pos_under).name
if minetest.get_item_group(node_above, "wall") == 1 then
	minetest.set_node(pos, {name = wall_name .. "_full"})
end

```

Reply:
I will swap out in favor of this suggestion. Should be done in the next day or so